language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/K8sCompletionRouter.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    private Map<String, CompletionItem> getTableCompletions(TableNode tableNode,
+                                                            Map<Parent, Map<String, CompletionItem>> snippets) {
+
+        String tableKey = TomlSyntaxTreeUtil.toDottedString(tableNode.identifier());
+        Map<String, CompletionItem> completions = new HashMap<>(getChildFromDottedKey(snippets, tableKey));
+
+        for (KeyValueNode field : tableNode.fields()) {
+            String key = TomlSyntaxTreeUtil.toDottedString(field.identifier());
+            completions.remove(key);
+        }
+        //Add main completions to completions
+        addRooTablesToCompletions(snippets.keySet(), completions);
+
+        //Remove Existing document level entries from completion
+        List<String> existingChildEntries = findExistingChildEntries(tableNode, completions);
+        return removeExistingChildTables(completions, existingChildEntries);
+    }
+
+    private Map<String, CompletionItem> addRooTablesToCompletions(Set<Parent> parents,
+                                                                  Map<String, CompletionItem> completions) {
+        for (Parent parent : parents) {
+            String key = parent.getKey();
+            if (completions.containsKey(key)) {
+                continue;
+            }
+            ParentType type = parent.getType();
+            CompletionItem item = new CompletionItem();
+            if (type == ParentType.TABLE) {
+                Table objectNode = new Table(key);
+                item.setInsertText(objectNode.prettyPrint());
+                item.setDetail(TomlSyntaxTreeUtil.TABLE);
+                item.setLabel(key);
+                item.setKind(CompletionItemKind.Snippet);
+                item.setInsertTextFormat(InsertTextFormat.Snippet);
+                item.setSortText(SortingUtil.genSortText(3));
+            } else if (type == ParentType.TABLE_ARRAY) {
+                TableArray objectNode = new TableArray(key);
+                item.setInsertText(objectNode.prettyPrint());
+                item.setLabel(key);
+                item.setKind(CompletionItemKind.Snippet);
+                item.setInsertTextFormat(InsertTextFormat.Snippet);
+                item.setDetail(TomlSyntaxTreeUtil.TABLE_ARRAY);
+                item.setSortText(SortingUtil.genSortText(3));
+            } else {
+                continue;
+            }
+            completions.put(key, item);
+        }
+        return completions;
+    }
+
+    private Map<String, CompletionItem> getChildFromDottedKey(Map<Parent, Map<String, CompletionItem>> snippets,
+                                                              String tableKey) {
+        for (Map.Entry<Parent, Map<String, CompletionItem>> entry : snippets.entrySet()) {
+            Parent key = entry.getKey();
+            if (key.getKey().equals(tableKey)) {
+                return entry.getValue();
+            }
+        }
+        return null;
```

Comment:
Instead of null, what if we return an empty map? Let's try to avoid nulls as much as possible